zenoh/src/net/routing/interceptor/access_control.rs

```diff
@@ -0,0 +1,337 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+//! ⚠️ WARNING ⚠️
+//!
+//! This module is intended for Zenoh's internal use.
+//!
+//! [Click here for Zenoh's documentation](../zenoh/index.html)
+
+use super::{
+    authorization::PolicyEnforcer, EgressInterceptor, IngressInterceptor, InterceptorFactory,
+    InterceptorFactoryTrait, InterceptorTrait,
+};
+use crate::net::routing::RoutingContext;
+use crate::KeyExpr;
+use std::any::Any;
+use std::sync::Arc;
+use zenoh_config::{AclConfig, Action, InterceptorFlow, Permission, Subject, ZenohId};
+use zenoh_protocol::{
+    network::{Declare, DeclareBody, NetworkBody, NetworkMessage, Push, Request},
+    zenoh::{PushBody, RequestBody},
+};
+use zenoh_result::ZResult;
+use zenoh_transport::{multicast::TransportMulticast, unicast::TransportUnicast};
+pub struct AclEnforcer {
+    enforcer: Arc<PolicyEnforcer>,
+}
+struct EgressAclEnforcer {
+    policy_enforcer: Arc<PolicyEnforcer>,
+    interface_list: Vec<i32>,
+    zid: ZenohId,
+}
+struct IngressAclEnforcer {
+    policy_enforcer: Arc<PolicyEnforcer>,
+    interface_list: Vec<i32>,
+    zid: ZenohId,
+}
+
+pub(crate) fn acl_interceptor_factories(
+    acl_config: &AclConfig,
+) -> ZResult<Vec<InterceptorFactory>> {
+    let mut res: Vec<InterceptorFactory> = vec![];
+
+    if acl_config.enabled {
+        let mut policy_enforcer = PolicyEnforcer::new();
+        match policy_enforcer.init(acl_config) {
+            Ok(_) => {
+                tracing::info!("[ACCESS LOG]: Access control is enabled and initialized");
+                res.push(Box::new(AclEnforcer {
+                    enforcer: Arc::new(policy_enforcer),
+                }))
+            }
+            Err(e) => tracing::error!(
+                "[ACCESS LOG]: Access control enabled but not initialized with error {}!",
```

Comment:
Adding `"[ACCESS LOG]"` makes the log look very weird when mixed with all the other zenoh logs. I suggest removing it until a broader zenoh log rework.